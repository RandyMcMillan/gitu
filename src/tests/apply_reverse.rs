use super::*;

#[test]
fn apply_hunk_no_newline() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "file.txt", "bad text");
    run(ctx.dir.path(), &["git", "checkout", "-b", "other"]);
    commit(ctx.dir.path(), "file.txt", "good text");
    run(ctx.dir.path(), &["git", "checkout", "main"]);

    snapshot!(ctx, "loother<enter><enter>aqq");
}

#[test]
fn apply_hunk() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "file.txt", "bad text\n");
    run(ctx.dir.path(), &["git", "checkout", "-b", "other"]);
    commit(ctx.dir.path(), "file.txt", "good text\n");
    run(ctx.dir.path(), &["git", "checkout", "main"]);

    snapshot!(ctx, "loother<enter><enter>aqq");
}

#[test]
fn apply_line() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "file.txt", "bad text\nmore bad text\n");
    run(ctx.dir.path(), &["git", "checkout", "-b", "other"]);
    commit(ctx.dir.path(), "file.txt", "good text\nmore good text\n");
    run(ctx.dir.path(), &["git", "checkout", "main"]);

    snapshot!(ctx, "loother<enter><enter><ctrl+j>a");
}

#[test]
fn reverse_line() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "file.txt", "bad text\nmore bad text\n");

    snapshot!(ctx, "ll<enter><ctrl+j>v");
}
