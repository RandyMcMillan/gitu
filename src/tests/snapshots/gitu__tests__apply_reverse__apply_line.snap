---
source: src/tests/apply_reverse.rs
expression: ctx.redact_buffer()
---
 commit b90fb319f35137259d898bedd7a0e01ac013c208                                |
 Author: Author Name <author@email.com>                                         |
 Date:   Fri, 16 Feb 2024 11:11:00 +0100                                        |
                                                                                |
     modify file.txt                                                            |
                                                                                |
     Commit body goes here                                                      |
                                                                                |
 modified   file.txt                                                            |
 @@ -1,2 +1,2 @@                                                                |
▌-bad text                                                                      |
 -more bad text                                                                 |
 +good text                                                                     |
 +more good text                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
$ git apply --recount                                                           |
styles_hash: b9817e6d642d48f2
